fv3core/stencils/yppm.py

```diff
@@ -229,306 +120,211 @@ def xt_dya_edge_1_base(q, dya):
+    else:
+        # effect of dm=0 included here
+        al = 0.0
+        ar = 0.0
+    return al, ar
 
 
 @gtstencil()
-def pert_ppm_standard_constraint(a0: sd, al: sd, ar: sd):
+def pert_ppm_standard_constraint(a0: FloatField, al: FloatField, ar: FloatField):
     with computation(PARALLEL), interval(...):
-        da1 = 0.0
-        da2 = 0.0
-        a6da = 0.0
-        if al * ar < 0.0:
-            da1 = al - ar
-            da2 = da1 ** 2
-            a6da = 3.0 * (al + ar) * da1
-            if a6da < -da2:
-                ar = -2.0 * al
-            elif a6da > da2:
-                al = -2.0 * ar
-        else:
-            # effect of dm=0 included here
-            al = 0.0
-            ar = 0.0
+        al, ar = pert_ppm_standard_constraint_fcn(a0, al, ar)
 
 
-def compute_al(q, dyvar, jord, ifirst, ilast, js1, je3, kstart=0, nk=None):
-    if nk is None:
-        nk = grid().npz - kstart
-    dimensions = q.shape
-    local_origin = (origin[0], origin[1], kstart)
-    al = utils.make_storage_from_shape(dimensions, local_origin)
-    if jord < 8:
-        main_al_ord_under8(
-            q,
-            al,
-            origin=(ifirst, js1, kstart),
-            domain=(ilast - ifirst + 1, je3 - js1 + 1, nk),
-        )
-        x_edge_domain = (dimensions[0], 1, nk)
-        if not grid().nested and spec.namelist.grid_type < 3:
-            # South Edge
-            if grid().south_edge:
-                al_x_under8_edge_0(
-                    q,
-                    dyvar,
-                    al,
-                    origin=(0, grid().js - 1, kstart),
-                    domain=x_edge_domain,
-                )
-                al_x_under8_edge_1(
-                    q, dyvar, al, origin=(0, grid().js, kstart), domain=x_edge_domain
-                )
-                al_x_under8_edge_2(
-                    q,
-                    dyvar,
-                    al,
-                    origin=(0, grid().js + 1, kstart),
-                    domain=x_edge_domain,
-                )
-            # North Edge
-            if grid().north_edge:
-                al_x_under8_edge_0(
-                    q, dyvar, al, origin=(0, grid().je, kstart), domain=x_edge_domain
-                )
-                al_x_under8_edge_1(
-                    q,
-                    dyvar,
-                    al,
-                    origin=(0, grid().je + 1, kstart),
-                    domain=x_edge_domain,
-                )
-                al_x_under8_edge_2(
-                    q,
-                    dyvar,
-                    al,
-                    origin=(0, grid().je + 2, kstart),
-                    domain=x_edge_domain,
-                )
-        if jord < 0:
-            floor_cap(
-                al,
-                0.0,
-                origin=(ifirst, grid().js - 1, kstart),
-                domain=(ilast - ifirst + 1, grid().njc + 3, nk),
-            )
+def compute_al(q: FloatField, dya: FloatField):
```

Comment:
If you understand what al is, I'd really appreciate seeing it in a docstring.

Reply:
Done!